CodeGeneration/Sourcery/Fakes/Fakes.swifttemplate

```diff
@@ -53,7 +53,7 @@ let specsToGenerate: [FakeableSpec] = matchingTypes.map { type in
 
     /// If we are generating for an `Enum`, return with `.value` for content 
     if let enumType = type as? Enum, let firstEnumCase = enumType.cases.first {
-        return FakeableSpec(name: enumType.name, accessLevelWithSpacePostfix: accessLevel, content: .value(firstEnumCase.name))
+        return FakeableSpec(name: enumType.globalName, accessLevelWithSpacePostfix: accessLevel, content: .value(firstEnumCase.name))
```

Comment:
I don't fully understand this bit, let me know if I'm off: 

We're changing the name property in the `FakeableSpec` template for `enumType.globalName`, which I understand would not take only the type name anymore , but the fully-qualified name (module name + type name + ... ) making these globally unique since now we have this sort of namespace assuring that there's no conflicts when there's more than one Enum named the same in different modules.

If I understood this correctly, does it means that we weren't doing this only for Enums? Since `globalName` seems to be used on the method return on line 72 here:
```
return FakeableSpec(name: type.globalName, accessLevelWithSpacePostfix: accessLevel, content: .initializer(propSpecs))
```

Reply:
@iamgabrielma Yep, all correct. If you look at the `Fakes/Hardware.generated` file as an example, before this change the first couple of generated examples are:

```
extension CardBrand {
    /// Returns a "ready to use" type filled with fake values.
    ///
    public static func fake() -> CardBrand {
        .visa
    }
}
extension Hardware.CardPresentReceiptParameters {
    /// Returns a "ready to use" type filled with fake values.
    ///
    public static func fake() -> Hardware.CardPresentReceiptParameters {
        .init(
            amount: .fake(),
            formattedAmount: .fake(),
            currency: .fake(),
            date: .fake(),
            storeName: .fake(),
            cardDetails: .fake(),
            orderID: .fake()
        )
    }
}
```

And you can see that the `CardPresentTransactionDetails` struct is fully qualified with the `Hardware` module prefix, but the `CardBrand` enum isn't.

This change gets us treating enums the same as other object types.